### PR TITLE
destination bigquery 1s1t: print subquery stats

### PR DESCRIPTION
**Pull request by edgao**
turns out it _is_ possible to to find the cost of each query within a transaction, it's just a little involved.

the bq.listJobs is a real api call, so it takes around 1 second ish. Arguably that shouldn't happen on the main thread, but... (1) meh, it's 1 additional second every few minutes, and (2) async should make this less bad since other threads are still doing things.

logs look something like this
image

also use this in the sqlgenerator integration test for convenience.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java [hunk 1]
```diff
@@ -47,13 +49,45 @@ public void execute(final String sql) throws InterruptedException {
...
+    // SQL transactions can spawn child jobs, which are billed individually. Log their stats too.
+    if (statistics.getNumChildJobs() != null) {
+      // There isn't (afaict) anything resembling job.getChildJobs(), so we have to ask bq for them
+      bq.listJobs(BigQuery.JobListOption.parentJobId(job.getJobId().getJob())).streamAll()
+          .sorted(Comparator.comparing(childJob -> childJob.getStatistics().getEndTime()))
+          .forEach(childJob -> {
+            JobConfiguration configuration = childJob.getConfiguration();
+            if (configuration instanceof QueryJobConfiguration qc) {
+              JobStatistics.QueryStatistics childQueryStats = childJob.getStatistics();
+              String truncatedQuery = qc.getQuery()
```
Comment (cynthiaxyin) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java:73]: nit: maybe add "..." to the end of the string if we truncate for clarity?

**Review by cynthiaxyin — APPROVED** [review]
This is cool!

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigQueryDestinationHandler.java:73]: done https://github.com/airbytehq/airbyte/pull/28373/commits/b283cae6bc897a90c663af9853a1c073c81c484d

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`81cb72af47`](https://github.com/airbytehq/airbyte/pull/28373/commits/81cb72af4781325a8e223ba7af792d16118465d8)) - ❌

⏲️  Total pipeline duration: 41mn09s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edga_1s1t_even_more_stats/1689634655/81cb72af4781325a8e223ba7af792d16118465d8/destination-bigquery/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`81cb72af47`](https://github.com/airbytehq/airbyte/pull/28373/commits/81cb72af4781325a8e223ba7af792d16118465d8)) - ✅

⏲️  Total pipeline duration: 22mn55s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edga_1s1t_even_more_stats/1689634655/81cb72af4781325a8e223ba7af792d16118465d8/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`126c6de791`](https://github.com/airbytehq/airbyte/pull/28373/commits/126c6de791aff74752b24d22b3dc0e1b4ad767a9)) - ✅

⏲️  Total pipeline duration: 24mn24s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edga_1s1t_even_more_stats/1689638630/126c6de791aff74752b24d22b3dc0e1b4ad767a9/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`126c6de791`](https://github.com/airbytehq/airbyte/pull/28373/commits/126c6de791aff74752b24d22b3dc0e1b4ad767a9)) - ✅

⏲️  Total pipeline duration: 37mn51s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edga_1s1t_even_more_stats/1689638630/126c6de791aff74752b24d22b3dc0e1b4ad767a9/destination-bigquery/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-approvington** [comment]
This looks fine!  <br/> Merged! <br/> Imagine it being fine

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**